crates/subspace-networking/src/connected_peers.rs

```diff
@@ -140,36 +153,29 @@ impl Behaviour {
             dialing_delay,
             peer_cache: Vec::new(),
             waker: None,
-            peer_connection_counters: HashMap::new(),
         }
     }
 
     /// Create a connection handler for the protocol.
     fn new_connection_handler(&mut self, peer_id: &PeerId) -> Handler {
         let default_until = Instant::now().add(self.config.decision_timeout);
         let default_keep_alive_until = KeepAlive::Until(default_until);
-        let keep_alive = if let Some(decision) = self.known_peers.get_mut(peer_id) {
-            match decision {
-                PeerConnectionState::Connecting { .. } => {
+        let keep_alive = if let Some(state) = self.known_peers.get_mut(peer_id) {
+            match state.connection_state {
+                ConnectionState::Connecting { .. } => {
                     // Connection attempt was successful.
-                    *decision = PeerConnectionState::Deciding {
-                        until: default_until,
-                    };
+                    state.connection_state = ConnectionState::Deciding;
 
                     default_keep_alive_until
                 }
-                PeerConnectionState::Deciding { until } => KeepAlive::Until(*until),
-                PeerConnectionState::Permanent => KeepAlive::Yes,
-                PeerConnectionState::NotInterested => KeepAlive::No,
+                ConnectionState::Deciding => KeepAlive::Until(default_until),
```

Comment:
I thought we discussed that `KeepAlive::Until()` is unnecessary here because other protocols might have it and generally connection timeout is still applicable, so no reason for this protocol to have additional `KeepAlive` until we know the decision.

I think we should return `KeepAlive::No` here and change it to `KeepAlive::Yes` later if necessary.

Reply:
We discussed that topic indeed, however, IIRC we had a different accent on this feature. On the first hand - other protocols could handle keep-alive for us, on the other hand, we need to have a decent number of connections for L1 lookups and we don't want to skip extra connection with a farmer because of KeepAlive::No as the default. In the end, we concluded that the timeout in the `PeerState::Deciding` is redundant and I removed it.

Reply:
That was a long comment, but I feel like it was answering some different question. So far I don't understand why it is necessary or beneficial in any way to have `KeepAlive::Until` here. While it is clear why not having extra parameter (`self.config.decision_timeout`) is obviously better.